datalake/src/main/java/com/sequenceiq/datalake/service/sdx/start/SdxStartService.java

```diff
@@ -0,0 +1,142 @@
+import com.sequenceiq.datalake.flow.SdxReactorFlowManager;
+import com.sequenceiq.datalake.flow.statestore.DatalakeInMemoryStateStore;
+import com.sequenceiq.datalake.service.sdx.PollingConfig;
+import com.sequenceiq.datalake.service.sdx.SdxNotificationService;
+import com.sequenceiq.datalake.service.sdx.SdxService;
+import com.sequenceiq.datalake.service.sdx.status.SdxStatusService;
+
+@Component
+public class SdxStartService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SdxStartService.class);
+
+    @Inject
+    private SdxReactorFlowManager sdxReactorFlowManager;
+
+    @Inject
+    private SdxService sdxService;
+
+    @Inject
+    private StackV4Endpoint stackV4Endpoint;
+
+    @Inject
+    private SdxNotificationService notificationService;
+
+    @Inject
+    private SdxStatusService sdxStatusService;
+
+    public void triggerStart(SdxCluster cluster) {
+        MDCBuilder.buildMdcContext(cluster);
+        sdxReactorFlowManager.triggerSdxStartFlow(cluster.getId());
+    }
+
+    public void start(Long sdxId) {
+        SdxCluster sdxCluster = sdxService.getById(sdxId);
+        try {
+            LOGGER.info("Triggering start flow for cluster {}", sdxCluster.getClusterName());
+            stackV4Endpoint.putStart(0L, sdxCluster.getClusterName());
+            notificationService.send(ResourceEvent.SDX_START_STARTED, sdxCluster);
+            sdxStatusService.setStatusForDatalake(DatalakeStatusEnum.START_IN_PROGRESS, "Datalake start in progress", sdxCluster);
+        } catch (NotFoundException e) {
+            LOGGER.info("Can not find stack on cloudbreak side {}", sdxCluster.getClusterName());
+        } catch (ClientErrorException e) {
+            String errorMessage = ClientErrorExceptionHandler.getErrorMessage(e);
+            LOGGER.info("Can not start stack {} from cloudbreak: {}", sdxCluster.getStackId(), errorMessage, e);
+            throw new RuntimeException("Can not start stack, client error happened on Cloudbreak side: " + errorMessage);
+        } catch (WebApplicationException e) {
+            LOGGER.info("Can not start stack {} from cloudbreak: {}", sdxCluster.getStackId(), e.getMessage(), e);
+            throw new RuntimeException("Can not start stack, client error happened on Cloudbreak side: " + e.getMessage());
+        }
+    }
+
+    public void waitCloudbreakCluster(Long sdxId, PollingConfig pollingConfig) {
+        SdxCluster sdxCluster = sdxService.getById(sdxId);
+        Polling.waitPeriodly(pollingConfig.getSleepTime(), pollingConfig.getSleepTimeUnit())
+                .stopIfException(pollingConfig.getStopPollingIfExceptionOccured())
+                .stopAfterDelay(pollingConfig.getDuration(), pollingConfig.getDurationTimeUnit())
+                .run(() -> checkClusterStatusDuringStart(sdxCluster));
+        sdxStatusService.setStatusForDatalake(DatalakeStatusEnum.RUNNING, "Datalake is running", sdxCluster);
+        sdxService.save(sdxCluster);
```

Comment:
I'm not sure if this repository save is needed here, status is now handled as a separate entity.